ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedWindowTableIQv2Test.java

```diff
@@ -0,0 +1,321 @@
+
+  @Mock
+  private KafkaStreams kafkaStreams;
+  @Mock
+  private KsStateStore stateStore;
+  @Mock
+  private WindowStoreIterator<ValueAndTimestamp<GenericRow>> fetchIterator;
+  @Mock
+  private KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>> keyValueIterator;
+  @Mock
+  private QueryResult<WindowStoreIterator<ValueAndTimestamp<GenericRow>>> fetchResult;
+  @Mock
+  private QueryResult<KeyValueIterator<Windowed<GenericKey>, ValueAndTimestamp<GenericRow>>> keyValueResult;
+
+  private KsMaterializedWindowTableIQv2 table;
+
+  @Before
+  public void setUp() {
+    table = new KsMaterializedWindowTableIQv2(stateStore, WINDOW_SIZE);
+
+    when(stateStore.getStateStoreName()).thenReturn("rocksdb-store");
+    when(stateStore.getKafkaStreams()).thenReturn(kafkaStreams);
+    when(stateStore.schema()).thenReturn(SCHEMA);
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Test
+  public void shouldThrowNPEs() {
+    new NullPointerTester()
+        .setDefault(KsStateStore.class, stateStore)
+        .testConstructors(KsMaterializedWindowTable.class, Visibility.PACKAGE);
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails() {
+    // Given:
+    when(kafkaStreams.query(any())).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFailedResult() {
+    // Given:
+    final StateQueryResult partitionResult = new StateQueryResult();
+    partitionResult.addResult(PARTITION, QueryResult.forFailure(FailureReason.STORE_EXCEPTION, "Boom"));
+    when(kafkaStreams.query(any())).thenReturn(partitionResult);
+
+    // When:
+    final Exception e = assertThrows(
+      MaterializationException.class,
+      () -> table.get(A_KEY, PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+      "Failed to get value from materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationException.class)));
+  }
+
+  @Test
+  public void shouldThrowIfGettingStateStoreFails_fetchAll() {
+    // Given:
+    when(stateStore.getKafkaStreams()).thenThrow(new MaterializationTimeOutException("Boom"));
+
+    // When:
+    final Exception e = assertThrows(
+        MaterializationException.class,
+        () -> table.get(PARTITION, WINDOW_START_BOUNDS, WINDOW_END_BOUNDS)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Failed to scan materialized table"));
+    assertThat(e.getCause(), (instanceOf(MaterializationTimeOutException.class)));
+  }
```

Comment:
I don't think we need this test (plus, it's not testing what it says it is).

I.e., it's testing what happens if `stateStore.getKafkaStreams()` throws an exception, which will never happen (because that method is just a field getter).

Reply:
Changed this to a case where the query fails